File: tomiServer/src/main/java/ca/projectTOMi/tomi/viewModel/BillableHoursReportLine.java
package ca.projectTOMi.tomi.viewModel;

import ca.projectTOMi.tomi.model.Project;
import lombok.Data;

/**
 * A single line from a Report to quickly analyse projects based on time spent on billable tasks.
 *
 * @author Karol Talbot
 * @version 1
 */
@Data
public class BillableHoursReportLine {
	/**
	 * The number of hours spent on billable tasks.
	 */
	private double billableHours;

	/**
	 * The number of hours spent on non billable tasks.
	 */
	private double nonbillableHours;

	/**
	 * The project being analysed.
	 */
	private Project project;

	/**
	 * Creates a new BillableHoursReportLine.
	 *
	 * @param billableHours
	 * 	Number of billable hours for the project
	 * @param nonbillableHours
	 * 	Number of non billable hours for the project
	 * @param project
	 * 	The project associated with the line
	 */
	public BillableHoursReportLine(final double billableHours, final double nonbillableHours, final Project project) {
		this.billableHours = billableHours;
		this.nonbillableHours = nonbillableHours;
		this.project = project;
	}
}

File: tomiClient/src/app/model/billableHoursReportLine.ts
import {Project} from "./project";

/**
 * A BillableHoursReportLine represents one line of a billable hour report.
 *
 * @author Jessica Moffatt
 * @version 1.0
 */
export class BillableHoursReportLine{
  /** The billable hours for the Project.*/
  billableHours:number;
  /** The non billable hours for the Project.*/
  nonbillableHours:number;
  /** The Project this billable hour report is for.*/
  project:Project;

  constructor(){
    this.billableHours = 0;
    this.nonbillableHours= 0;
    this.project = null;
  }

  /**
   * Method used to compare two BillableHoursReportLines by the Project's name.
   * @param first The first BillableHoursReportLine to compare.
   * @param second The second BillableHoursReportLine to compare.
   */
  public static compareName(first: BillableHoursReportLine, second:BillableHoursReportLine){
    if (first.project.projectName < second.project.projectName) {
      return -1;
    }
    else if (first.project.projectName > second.project.projectName)
    {
      return 1;
    }
    return 0;
  }
}
